packages/core/src/connectors/aliyun-sms/index.ts

```diff
@@ -0,0 +1,88 @@
+import { z } from 'zod';
+
+import {
+  ConnectorConfigError,
+  ConnectorError,
+  ConnectorMetadata,
+  ConnectorType,
+  TextSendMessageFunction,
+  ValidateConfig,
+} from '../types';
+import { getConnectorConfig } from '../utilities';
+import { singleSendText } from './single-send-text';
+
+export const metadata: ConnectorMetadata = {
+  id: 'aliyun-sms',
+  type: ConnectorType.SMS,
+  name: {
+    en: 'Aliyun Short Message Service',
+    zh_CN: '阿里云短信服务',
+  },
+  logo: './logo.png',
+  description: {
+    en: 'Short Message Service (SMS) has a batch sending feature and various API operations to send one-time password (OTP) messages, notification messages, and promotional messages to customers.',
+    zh_CN:
+      '短信服务（Short Message Service）是指通过调用短信发送API，将指定短信内容发送给指定手机用户。',
+  },
+};
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  if (!config) {
+    throw new ConnectorConfigError('Missing config');
+  }
+
+  const result = configGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorConfigError(result.error.message);
+  }
+};
+
+export enum TemplateType {
+  Passcode = 0,
+  Notification = 1,
+  Promotion = 2,
+  InternationalMessage = 3,
+}
+
+const configGuard = z.object({
+  accessKeyId: z.string(),
+  accessKeySecret: z.string(),
+  signName: z.string(),
+  templates: z.array(
+    z.object({
+      type: z.nativeEnum(TemplateType).default(0),
+      code: z.string().optional(),
+      name: z.string().min(1).max(30),
+      content: z.string().min(1).max(500),
+      remark: z.string(),
+    })
+  ),
+});
+
+export type AliyunSmsConfig = z.infer<typeof configGuard>;
+
+export const sendMessage: TextSendMessageFunction = async (
+  phone,
+  signName,
+  templateCode,
+  payload
+) => {
+  const config: AliyunSmsConfig = await getConnectorConfig<AliyunSmsConfig>(metadata.id);
```

Comment:
No need to specify the config type again should be able to infer from the return type.